lib/worker_status.rb

```diff
@@ -0,0 +1,41 @@
+if Rails.configuration.active_job.queue_adapter == :delayed_job
```

Comment:
It seems like it could be just a conditional instantiation of one of two different (sub)classes. I don't think the code evaluation cost is a big deal?